Lib/test/test_netrc.py

```diff
@@ -123,6 +124,33 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
+            self.assertRaises(FileNotFoundError, netrc.netrc)
+
+    def test_file_not_found_explicit(self):
+        self.assertRaises(FileNotFoundError, netrc.netrc,
+                          file="unlikely_netrc")
+
+    def test_home_not_set(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        fake_netrc_path = os.path.join(d, 'netrc')
+        with open(fake_netrc_path, 'w') as f:
+            f.write("machine foo.domain.com login bar password pass")
+        os.chmod(fake_netrc_path, 0o600)
+        with support.EnvironmentVarGuard() as environ:
+            environ.unset('HOME')
+            with mock.patch('os.path.expanduser') as mock_expanduser:
```

Comment:
This is too fragile. If rather than `os.path.expanduser("~/.netrc")` use one of variants suggested by me, the test will be broken.

I would monkey-patch expanduser with a code like the following:
```python
orig_expanduser = os.path.expanduser
called = []
def fake_expanduser(s):
    called.append(s)
    environ.set('HOME', d)
    result = orig_expanduser(s)
    environ.unset('HOME')
    return result
with test.support.swap_attr(os.path, 'expanduser', fake_expanduser):
    nrc = netrc.netrc()
self.assertTrue(called)
```